config.m4

```diff
@@ -10,6 +11,11 @@ if test "$PHP_DDTRACE" != "no"; then
     PHP_SUBST(EXTRA_CFLAGS)
   fi
 
+  if test "$PHP_DDTRACE_DEBUG" != "no"; then
+    AC_DEFINE(DDTRACE_DEBUG, 1, [ ])
+    STD_CFLAGS="-g -O0 -Wall"
```

Comment:
I believe most of debug mode enabling will happen when the extension is checked out. As such its probably more useful to have the debug mode in main Makefile since its used more often, and also allows to easily switch builds between versions due to out of tree builds.

Reply:
TBH I'm not a big fan the enable/disable flags of pecl (they are very hard to automate when installing etc.)

Reply:
> I believe most of debug mode enabling will happen when the extension is checked out. 

Agreed.

> As such its probably more useful to have the debug mode in main Makefile since its used more often

We don't have any documentation around the Makefile so when an extension developer does a git checkout, they will likely run the common extension building commands (`$ phpize && ./configure && make && sudo make install`) which blows away the packaged Makefile anyway. :) What do you think about migrating away from the packaged Makefile by moving some features to the config script (like debug mode) and other features to a makefile fragment (like extra make targets). That way we follow more conventional extension development practices.  What are your thoughts? :)

Reply:
I think `phpize` is broken in that it overwrites Makefile and doesn't support out of folder builds 
> We don't have any documentation around the Makefile 
 we can add some :)

We did previously use another Makefile that wasn't the best experience to always type `make -f DDMakefile test_c` - esp since simple make without `-f  DDMakefile` was almost never used

> What do you think about migrating away from the packaged Makefile by moving some features to the config script (like debug mode) and other features to a makefile fragment (like extra make targets). That way we follow more conventional extension development practices. What are your thoughts? :) 

Makefile has good builtin dependency management which both scripts and composer doesn't have.

For most of the tasks within the current Makefile its quite crucial to have dependency management to have speedy builds.

The main purpose of current Makefile is to allow executing things like:
```
docker-compose run 7.2 make test_c
docker-compose run 7.1 make test_c
docker-compose run 7.0 make test_c
docker-compose run 5.6 make test_c
docker-compose run 5.4 make test_c
```

And  don't have to care about phpize leftovers between versions and have fast recompilation on changes

But this gives me idea that I think adding a script like `bin/make` which would:
```bash
#!/bin/bash
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"

make -f $DIR/../Makefile.something  "$@"
```

Would probably solve most problems with adding `-f DDMakefile` to every make invocation.